### PR TITLE
Ggiraph demo

### PR DESCRIPTION
**Pull request by jr-nicola**
- using .rds files of saved ggplot2 trees instead of saved htmlwidgets
- Load rds files into app and convert to ggiraph output
- Record click data and display cluster clicked on as text
- Use click on graph to choose plot/table/rds
- Remove drop down menu as ggtree plot now works as selector instead
- Have removed old .html files as no longer required in the app (since .rds files used instead)
- Have removed plot_sina outputs from inst folder for now, as not currently included in the app
- Now have 1 note in R CMD check resulting from file size (unavoidable for this app)

Note: included a hacky workaround to process the tooltip data and extract the cluster_id based on the value of data_id (if we reduce the tooltip to just having the cluster id as planned, then this code should simplify a lot more and we can just use `parse_number()` instead)

### CODE REVIEW

File: R/utils.R [hunk 1]
```diff
@@ -2,11 +2,14 @@
 #' @param type Character string detailing type of widget to show
 get_filename = function(type) {
   filename = switch(type,
-                    "Logistic growth rate" = "tree-logistic_growth_rate-2021-11-27.html", # nolint
-                    "Simple logistic growth rate" = "tree-simple_logistic_growth_rate-2021-11-27.html", # nolint
-                    "Simple trait log odds" = "tree-sim_trait_logodds-2021-11-27.html" # nolint
+                    "Logistic growth rate" = "tree-logistic_growth_rate-2021-11-27.rds", # nolint
+                    "Simple logistic growth rate" = "tree-simple_logistic_growth_rate-2021-11-27.rds", # nolint
+                    "Simple trait log odds" = "tree-sim_trait_logodds-2021-11-27.rds" # nolint
```
Comment (russHyde) [inline R/utils.R:7]: Do you want the dates embedded inside the file paths here? The app source code would need updating whenever the treeview() function is reran. The original code for making the .html files made a copy of the most-recent file that was named "tree-logistic_growth_rate.html". We could replicate that to make it simpler to always use the most-recent charts here

Comment (jr-nicola) [reply R/utils.R:7]: I'm happy for this just to be "tree-logistic_growth_rate.html" since it will make it easier to update

**Review by russHyde — CHANGES_REQUESTED** [review]
It's mostly good. I _think_ there's an easier way to extract the cluster IDs from the tooltip strings.

I did have a look to see if the warning that is thrown on converting the ggtree to girafe is a known issue. There is a SO post with the same warning [here](https://stackoverflow.com/questions/74367811/error-in-ggqqplot-the-following-aesthetics-were-dropped-during-statistical-tran)

File: R/utils.R [hunk 2]
```diff
@@ -134,3 +137,45 @@ downloader_tab_panel = function(title,
     )
   )
 }
+
+#' function to get node id from data_id column of ggplot
+#' @param tooltip_input Character vector of tooltip content
+get_cluster_ID = function(tooltip_input) {
```
[the remark this answers is not in the document]
Comment (jr-nicola) [reply R/utils.R:143]: I've changed this over

File: tests/testthat/test-cluster-ids.R [hunk 3]
```diff
@@ -0,0 +1,6 @@
+test_that("cluster id is parsed from tooltip correctly", {
+  tooltip = c("Statistics:\n                                             \n------------------  -------------------------\nCluster.ID          #1605                    \nCluster.size        25                       \nDate.range          2021-01-01 -> 2021-06-21 \nExample.sequence    EPI_ISL_4817051", # nolint
+              "Statistics:\n                                             \n------------------  -------------------------\nCluster.ID          #1920                    \nCluster.size        65                       \nDate.range          2021-06-02 -> 2021-10-22 \nExample.sequence    EPI_ISL_6436444") # nolint
+  cluster_ids = tfpbrowser:::get_cluster_ID(tooltip)
```
Comment (russHyde) [inline tests/testthat/test-cluster-ids.R:4]: The `tfpbrowser:::` prefix needn't be present here, that'll fix the private-function lint.
```suggestion
  cluster_ids = get_cluster_ID(tooltip)
```

File: R/utils.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (russHyde) [reply R/utils.R:143]: Thanks for changing this

**Review by russHyde — APPROVED** [review]
I've approved. But I did add a suggestion to fix a lint in your test script.

File: tests/testthat/test-cluster-ids.R (hunk 3 above)
Comment (jr-nicola) [reply tests/testthat/test-cluster-ids.R:4]: Testthat failed when I removed the ::: so I've created an issue (#24 ) to fix it

**Pull request merged**